discord-bot/src/storage.ts

```diff
@@ -0,0 +1,68 @@
+import { createClient } from '@supabase/supabase-js'
+import { config } from './constants/config.js'
+import { User } from 'discord.js'
+import { Manifold } from 'manifold-sdk'
+
+const discordIdsToApiKeys: { [k: string]: string } = {}
+
+const { supabaseInstanceId } = config
+function getInstanceHostname(instanceId: string) {
+  return `${instanceId}.supabase.co`
+}
+
+function createSupabaseClient() {
+  const instanceId = supabaseInstanceId
+  if (!instanceId) {
+    throw new Error(
+      "Can't connect to Supabase; no process.env.SUPABASE_INSTANCE_ID and no instance ID in config."
+    )
+  }
+  const key = process.env.SUPABASE_KEY
+  if (!key) {
+    throw new Error("Can't connect to Supabase; no process.env.SUPABASE_KEY.")
+  }
+  const url = `https://${getInstanceHostname(instanceId)}`
+  return createClient(url, key)
+}
+
+export const supabase = createSupabaseClient()
```

Comment:
Seems like a lot of this stuff you could get from `common`. Note that `backend/shared` also has Supabase code.